Add docs for API v10 (#4927)
hammerandchisel/discord-api-docs: docs/Change_Log.md

# Change Log

## API v10

#### May 12, 2022

> info
> API v10 was made available in [Feb 2022](https://github.com/discord/discord-api-docs/discussions/4510). May 12 just indicates the date for this changelog entry.

- API v8 is now deprecated.
- `GET /channels/{channel.id}/threads/active` is decomissioned in favor of [`GET/guilds/{guild.id}/threads/active`](#DOCS_RESOURCES_GUILD/list-active-threads).

> - Starting in v10, you must specify the message content intent (using `1 << 15`). See more in the [Gateway Intents documentation](#DOCS_TOPICS_GATEWAY/gateway-intents).

Reviewer: ```suggestion
- Starting in v10, you must specify the message content intent (`1 << 15`) to receive content-related fields in message dispatches. See more in the [Gateway Intents documentation](#DOCS_TOPICS_GATEWAY/gateway-intents).
```

Reviewer 2: This should mention the affected fields rather than a general statement, as some fields like mentioned roles will still be there despite being part of the message content

> - To specify a reason for an administrative action in audit logs, apps must now pass the `X-Audit-Log-Reason` rather than the `reason` parameter. Read more in the [Audit Logs documentation](#DOCS_RESOURCES_AUDIT_LOG).

Reviewer 3: ```suggestion
- The `reason` field has been removed for `PUT /guilds/{guild.id}/bans/{user.id}` and `POST /guilds/{guild.id}/prune` endpoints. The `X-Audit-Log-Reason` header should be used instead. Read more in the [Audit Logs documentation](#DOCS_RESOURCES_AUDIT_LOG).
```

AFAIK the `reason` field parameter only existed for these 2 endpoints.
The original phrasing also looked like that `X-Audit-Log-Reason` was something new, but it has been around for years.

> - Message routes (like [`POST /channels/{channel.id}/messages`](#DOCS_RESOURCES_CHANNEL/create-message)) now use the `embeds` (an array of embed objects) instead of `embed`.

Reviewer 3: ```suggestion
- Message routes (like [`POST /channels/{channel.id}/messages`](#DOCS_RESOURCES_CHANNEL/create-message)) now use the `embeds` field (an array of embed objects) instead of `embed`.
```

Reviewer 3: btw, isn't this something that changed in v8?

- The `summary` field for [applications](#DOCS_RESOURCES_APPLICATION) now returns an empty string for all API versions.
- The `name` and `description` fields for [Achievements](#DOCS_GAME_SDK_ACHIEVEMENTS/data-models-achievement-struct) are now strings, and localization info is now passed in new `name_localizations` and `description_localizations` dictionaries. This change standardizes localization to match [Application Commands](#DOCS_INTERACTIONS_APPLICATION_COMMANDS/localization). Read details in the [Achievements documentation](#DOCS_GAME_SDK_ACHIEVEMENTS/data-models-achievement-struct).

> - Requests to v10 and higher will no longer be supported on `discordapp.com`ß (this does **not** affect `cdn.discordapp.com`)

Reviewer 4: ```suggestion
- Requests to v10 and higher will no longer be supported on `discordapp.com` (this does **not** affect `cdn.discordapp.com`)
```

> - Requests to v10 and higher will no longer be supported on `discordapp.com`ß (this does **not** affect `cdn.discordapp.com`)

Reviewer 3: ```suggestion
- Requests to v10 and higher will no longer be supported on `discordapp.com` (this does **not** affect `cdn.discordapp.com`)
```

> ## Upcoming changes

Reviewer 5: Could this block be moved up to be above the v10 changes? it will be more visible there keeps the changelog chronological

Author: Yeah I was debating between backdating the v10 entry date to February to add consistency, or to reflect the date of change log publishing. I'll do this and backdate though—agree it's better to keep releases chronological.

- API v10 will become the default version **in early 2023**
- API v6 and v7 will be decommissioned **in early 2023**

> - Message content is becoming a priviledged intent for verified bots in 75+ servers **on August 31, 2022**. Read details in [the FAQ](https://support-dev.discord.com/hc/en-us/articles/4404772028055-Message-Content-Privileged-Intent-FAQ) or follow [the guide](#DOCS_TUTORIALS_UPGRADING_TO_APPLICATION_COMMANDS) on updating your app.

Reviewer 6: peiviledged -> privileged

Author: 🙃

> - The `summary` field for applications will stop being sent in the next API version (v11)

Reviewer: ```suggestion
- The `summary` field for applications will be removed in the next API version (v11)
```
